packages/jest-util/src/__tests__/deep_cyclic_copy.test.js

```diff
@@ -70,9 +70,144 @@ it('uses the blacklist to avoid copying properties on the first level', () => {
     },
   };
 
-  expect(deepCyclicCopy(obj, new Set(['blacklisted']))).toEqual({
+  expect(deepCyclicCopy(obj, {blacklist: new Set(['blacklisted'])})).toEqual({
     subObj: {
       blacklisted: 42,
     },
   });
 });
+
+it('does not keep the prototype by default when top level is object', () => {
+  const sourceObject = new function() {}();
+  sourceObject.nestedObject = new function() {}();
+  sourceObject.nestedArray = new function() {
+    this.length = 0;
+  }();
+
+  const spy = jest.spyOn(Array, 'isArray').mockImplementation(object => {
+    return object === sourceObject.nestedArray;
+  });
+
+  const copy = deepCyclicCopy(sourceObject, {keepPrototype: false});
+
+  expect(Object.getPrototypeOf(copy)).not.toBe(
+    Object.getPrototypeOf(sourceObject),
+  );
+  expect(Object.getPrototypeOf(copy.nestedObject)).not.toBe(
+    Object.getPrototypeOf(sourceObject.nestedObject),
+  );
+  expect(Object.getPrototypeOf(copy.nestedArray)).not.toBe(
+    Object.getPrototypeOf(sourceObject.nestedArray),
+  );
+
+  expect(Object.getPrototypeOf(copy)).toBe(Object.getPrototypeOf({}));
+  expect(Object.getPrototypeOf(copy.nestedObject)).toBe(
+    Object.getPrototypeOf({}),
+  );
+  expect(Object.getPrototypeOf(copy.nestedArray)).toBe(
+    Object.getPrototypeOf([]),
+  );
+
+  spy.mockRestore();
+});
+
+it('does not keep the prototype by default when top level is array', () => {
+  const spy = jest.spyOn(Array, 'isArray').mockImplementation(() => true);
+
+  const sourceArray = new function() {
+    this.length = 0;
+  }();
+
+  const copy = deepCyclicCopy(sourceArray);
+  expect(Object.getPrototypeOf(copy)).not.toBe(
+    Object.getPrototypeOf(sourceArray),
+  );
+
+  expect(Object.getPrototypeOf(copy)).toBe(Object.getPrototypeOf([]));
+  spy.mockRestore();
+});
+
+it('does not keep the prototype of arrays when keepPrototype = false', () => {
```

Comment:
This test is a duplicate of above

Reply:
One asserts the default behavior and the other with the flag explicitly passed in

Do you think that's too much?

Reply:
I mean its content is literally a copy 😄 

<img width="629" alt="screen shot 2018-01-14 at 00 35 29" src="https://user-images.githubusercontent.com/5106466/34911110-f81a62f4-f8c2-11e7-8682-1169e64092df.png">

Reply:
Oh wow good catch -- the second test should have:

```js
const copy = deepCyclicCopy(sourceArray, {keepPrototype: false});
```